NameLayerMain/src/vg/civcraft/mc/namelayer/gui/AdminFunctionsGUI.java

```diff
@@ -0,0 +1,304 @@
+		}
+		ci.setSlot(deletionClick, 16);
+
+		// back button
+		ItemStack backToOverview = new ItemStack(Material.WOOD_DOOR);
+		ISUtils.setName(backToOverview, ChatColor.GOLD + "Back to overview");
+		ci.setSlot(new Clickable(backToOverview) {
+
+			@Override
+			public void clicked(Player arg0) {
+				parent.showScreen();
+			}
+		}, 22);
+		ci.showInventory(p);
+	}
+
+	private void showLinkingMenu() {
+		LinkingGUI lgui = new LinkingGUI(g, p, this);
+		lgui.showScreen();
+	}
+
+	private void showMergingMenu() {
+		MergeGUI mGui = new MergeGUI(g, p, this);
+		mGui.showScreen();
+	}
+
+	private void showTransferingMenu() {
+		p.sendMessage(ChatColor.GOLD
+				+ "Enter the name of the new primary owner or \"cancel\" to exit this prompt");
+		Dialog dia = new Dialog(p, NameLayerPlugin.getInstance()) {
+
+			@Override
+			public List<String> onTabComplete(String word, String[] arg1) {
+				List<String> names;
+				if (NameLayerPlugin.isMercuryEnabled()) {
+					names = new LinkedList<String>(MercuryAPI.getAllPlayers());
+				} else {
+					names = new LinkedList<String>();
+					for (Player p : Bukkit.getOnlinePlayers()) {
+						names.add(p.getName());
+					}
+				}
+				if (word.equals("")) {
+					return names;
+				}
+				List<String> result = new LinkedList<String>();
+				String comp = word.toLowerCase();
+				for (String s : names) {
+					if (s.toLowerCase().startsWith(comp)) {
+						result.add(s);
+					}
+				}
+				return result;
+			}
+
+			@Override
+			public void onReply(String[] arg0) {
+				if (arg0.length > 1) {
+					p.sendMessage(ChatColor.RED
+							+ "You may only enter one player to transfer to");
+					showScreen();
+					return;
+				}
+				if (arg0[0].equals("cancel")) {
+					showScreen();
+					return;
+				}
+				final UUID transferUUID = NameAPI.getUUID(arg0[0]);
+				if (transferUUID == null) {
+					p.sendMessage(ChatColor.RED + "This player doesn't exist");
+					showScreen();
+					return;
+				}
+				final String playerName = NameAPI.getCurrentName(transferUUID);
+				ClickableInventory confirmInv = new ClickableInventory(27,
+						g.getName());
+				ItemStack info = new ItemStack(Material.PAPER);
+				ISUtils.setName(info, ChatColor.GOLD + "Transfer group to "
+						+ playerName);
+				ISUtils.addLore(info, ChatColor.RED
+						+ "Are you sure that you want to", ChatColor.RED
+						+ "transfer this group? You can not undo this!");
+				ItemStack yes = new ItemStack(Material.INK_SACK);
+				yes.setDurability((short) 10); // green
+				ISUtils.setName(yes,
+						ChatColor.GOLD + "Yes, transfer	 " + g.getName()
+								+ " to " + playerName);
+				ItemStack no = new ItemStack(Material.INK_SACK);
+				no.setDurability((short) 1); // red
+				ISUtils.setName(no,
+						ChatColor.GOLD + "No, don't transfer " + g.getName());
+				confirmInv.setSlot(new Clickable(yes) {
+
+					@Override
+					public void clicked(Player p) {
+						if (!g.isOwner(p.getUniqueId())) {
```

Comment:
Well the issue is that we want different error handling for both versions and possibly different messages. Actually changing the owner is encapsulated, but all the checks need to be done beforehand.